source/sparse_matrix_simd.template.h

```diff
@@ -194,22 +194,51 @@ namespace ryujin
       send_targets.resize(partitioner->import_targets().size());
       auto idx = import_indices_part.begin();
 
+      /*
+       * Index p iterates over import_targets() and index p_match iterates
+       * over ghost_ranges. such that
+       *   import_targets()[p].first == ghost_rangs[p_match].first
+       */
+      unsigned int p_match = 0;
       for (unsigned int p = 0; p < partitioner->import_targets().size(); ++p) {
+
+        /*
+         * Match up processor index between receive and import targets. If
+         * we do not find a match, which can happen for locally refined
+         * meshes - then set p_match equal to receive_targests.size().
+         */
+        while (p_match < receive_targets.size() &&
+               receive_targets[p_match].first !=
+                   partitioner->import_targets()[p].first)
+          p_match++;
```

Comment:
Are we sure that we cannot increment `p_match` too quickly, i.e., we increment to the end `p_match = receive_targets.size()`, even though we would later have `p` such that we have a match? I have the feeling that it might work, but I am not entirely sure right now.

Reply:
This is under the assumption that the MPI ranks are sorted, which seems to be the case. But let me investigate.

Reply:
The MPI ranks in the list given by the Partitioner class are definitely sorted, which comes out from the `ComputeIndexOwner` namespace. I was just not sure whether gaps in the list could make us eagerly return to the end (more specifically, using `!=` rather than `<`).

Reply:
I understand. I hope the logic is fine - at least for everything I tested.